modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/EmbeddedScene.java

```diff
@@ -155,6 +155,7 @@ public void setPixelScaleFactors(float scalex, float scaley) {
         renderScaleX = scalex;
         renderScaleY = scaley;
         entireSceneNeedsRepaint();
+        sceneState.update();
```

Comment:
This is not thread-safe. This method must only be called on the JavaFX event thread, and with the render lock held. You will need a different solution to the problem you are trying to solve.

Reply:
ok I see in SWT this is called from the correct Thread but in Swing is is called from EDT. The problem is that sceneState update is ONLY called when the scene is dirty and there's not even internal API to call.

Reply:
even if called on the correct thread it will need the renderlock, and I don't know if this is an OK place to acquire that lock (it might be, but that would need to be explored).